.github/workflows/ci.yml

```diff
@@ -193,6 +193,38 @@ jobs:
           # Sadly 1 day is the minimum
           retention-days: 1
 
+  unit_tests_tsserver:
+    name: Run Unit Tests with Experimental TSServer
+    needs: [build]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package:
+          [
+            'eslint-plugin',
+            'eslint-plugin-internal',
+            'eslint-plugin-tslint',
+            'typescript-estree',
+          ]
+    env:
+      COLLECT_COVERAGE: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - name: Install
+        uses: ./.github/actions/prepare-install
+        with:
+          node-version: 18
+      - name: Build
+        uses: ./.github/actions/prepare-build
+      - name: Run unit tests for ${{ matrix.package }}
+        run: npx nx test ${{ matrix.package }} --coverage=false
+        env:
+          CI: true
+          TYPESCRIPT_ESLINT_EXPERIMENTAL_TSSERVER: true
```

Comment:
<img src="https://media4.giphy.com/media/cVVoIFsKoPkFq/giphy.gif"/>

This was a great thing to add!